rust/datafusion/src/execution/context.rs

```diff
@@ -987,22 +1009,23 @@ mod tests {
             Ok(Arc::new(add(l, r)?))
         });
 
-        let my_add = ScalarFunction::new(
```

Comment:
This is the main change to end users:

For most use-cases, `create_udf` is enough as it covers the common use-case of "single signature", "single return type".

For all other use-cases, the user can use `ScalarFunction::new`.